### DISCUSSION TITLE
Set thumbnail and featured image in front matter

### DISCUSSION THREAD

**Discussion by LITUATUI** [razonyang/hugo-theme-bootstrap#463] [Q&A]
Hi guys.

I'm migrating a big blog from WordPress to Hugo and my converted markdown posts have a featureImage field set in the front matter that I want to use both as thumbnail and as featured image (beginning of the post).

Is there a easy way to do this in this theme?

For example, in the [Mainroad theme](https://github.com/Vimux/Mainroad), If I set a `thumbnail: image.jpg` in the front matter, it's used as thumbnail and as featured image, just as I want. So I can use VSCode to batch search and replace the featureImage field with thumbnail in my markdown posts.

I'm dealing with hundreds of markdown files, it would be nice to know a easy/fast way to do this in this theme.

**Comment by razonyang** [comment]
Please take a look at the [Thumbnail Selection Order](https://hbs.razonyang.com/en/docs/content/#thumbnail-selection-order) if you haven't read it yet.

If your thumbnails are located at `static` folder or external platforms, the `images` parameter would be a good solution:

```toml
+++
images = [
  "/images/my-thumbnail.png" # Relative to `static` folder.
  # "https://example.com/images/foo.png" # External images.
]
+++
```

If the thumbnails are page resources, you'll need to move them to page level, and rename it to `feature.png`, `thumbnail.jpg` or `cover.png` etc.

**Reply by LITUATUI** [reply]
Thanks for the fast reply. Yes, did read that documentation.

I can batch search and replace the featureImage field in the front matter and get the thumbnails to show up, but what about the featured image - the image that appear at the beginning of the post? Can it be set in the front matter, or it is just for thumbnails?

**Reply by razonyang** [reply]
Currently, there is no such functionality that display a featured image at every top of posts. Could you please provide some real examples about this?

**Reply by LITUATUI** [reply]
Sure. 

The [blog](https://pushevs.com/) that I'm trying to migrate from WordPress to Hugo shows the behavior I want. Use the same image as thumbnail and featured (start of the post).

As I mentioned previously, some Hugo themes do this. 

For example, in the [Mainroad theme](https://github.com/Vimux/Mainroad), If I set a thumbnail: image.jpg in the front matter, it's used as thumbnail and as featured image, just as I want. So I can use VSCode to batch search and replace the featureImage field with thumbnail in my markdown posts.

This makes the migration to Hugo much simpler, since the most popular [plugin](https://github.com/SchumacherFM/wordpress-to-hugo-exporter) to convert WordPress posts to markdown generates the following fields in the front matter:

title:
author:
type:
date:
url: 
featureImage:
categories:
tags:

In the Mainroad theme I just batch search and replace `featureImage:` with `thumbnail:`.

**Reply by razonyang** [reply]
I see, but it won't be implemented like this, the current thumbnails look up order follows the same rules of Hugo built in [Twitter Card](https://github.com/gohugoio/hugo/blob/f6745ad3588a7b3aaae228fec18fe0027affd566/tpl/tplimpl/embedded/templates/twitter_cards.html) and [Open Graph](https://github.com/gohugoio/hugo/blob/f6745ad3588a7b3aaae228fec18fe0027affd566/tpl/tplimpl/embedded/templates/opengraph.html) templates. The `thumbnail` is not compatible with those templates.

As for the **featured** images at the beginning of posts, it's reasonable to display a featured image on a post. Could you please create a [feature request](https://github.com/razonyang/hugo-theme-bootstrap/issues/new?assignees=&labels=enhancement&template=feature_request.md&title=) and describe it in details?

**Reply by LITUATUI** [reply]
Thanks mate. Will do.
